src/models/set_tmp.js

```diff
@@ -0,0 +1,315 @@
+import {
+  units_converter,
+  units_converter_array,
+  check_standard_compliance,
+  check_standard_compliance_array,
+  round,
+} from "../utilities/utilities.js";
+import { two_nodes, two_nodes_array, roundArray } from "../models/two_nodes.js";
+
+/**
+ * @typedef {Object} SetTmpKwargs
+ * @property {boolean} [round=true]
+ * @property {boolean} [calculate_ce=false]
+ * @public
+ */
+
+/**
+ * @typedef {Object} SetTmpKwargsRequired
+ * @property {boolean} round
+ * @property {boolean} calculate_ce
+ */
+/**
+ * Calculates the Standard Effective Temperature (SET). The SET is the
+ * temperature of a hypothetical isothermal environment at 50% (rh),
+ * <0.1 m/s (20 fpm) average air speed (v), and tr = tdb, in which the
+ * total heat loss from the skin of an imaginary occupant wearing
+ * clothing, standardized for the activity concerned is the same as
+ * that from a person in the actual environment with actual clothing
+ * and activity level.
+ * @public
+ * @memberof models
+ * @docname Standard Effective Temperature (SET)
+ *
+ * @see {@link set_tmp_array} for a version that supports arrays
+ *
+ * @param {number} tdb Dry bulb air temperature, default in [°C] in [°F] if `units` = 'IP'.
+ * @param {number} tr Mean radiant temperature, default in [°C]
+ * @param {number} v Air speed, default in [m/s]
+ * @param {number} rh Relative humidity, [%].
+ * @param {number} met Metabolic rate, [W/(m2)]
+ * @param {number} clo Clothing insulation, [clo]
+ * @param {number} [wme=0] External work, [W/(m2)] default 0
+ * @param {number} [body_surface_area=1.8258] Body surface area, default value 1.8258 [m2] in [ft2] if units = ‘IP’
+ * @param {number} [p_atm=101325] Atmospheric pressure, default value 101325 [Pa] in [atm] if units = ‘IP’
+ * @param {"standing" | "sitting"} [body_position="standing"] Select either “sitting” or “standing”
+ * @param {"SI" | "IP"} [units="SI"] Select the SI (International System of Units) or the IP (Imperial Units) system.
+ * @param {boolean} [limit_inputs=true] By default, if the inputs are outsude the following limits the function returns nan. If False returns values regardless of the input values.
+ * @param {SetTmpKwargs} [kwargs]
+ * @returns {number} SET – Standard effective temperature in array, [°C]
+ *
+ * @example
+ * const set = set_tmp(25, 25, 0.1, 50, 1.2, 0.5); // returns 24.3
+ */
+export function set_tmp(
+  tdb,
+  tr,
+  v,
+  rh,
+  met,
+  clo,
+  wme = 0,
+  body_surface_area = 1.8258,
+  p_atm = 101325,
+  body_position = "standing",
+  units = "SI",
+  limit_inputs = true,
+  kwargs = {},
+) {
+  const defaults_kwargs = {
+    calculate_ce: false,
+    round: true,
+  };
+
+  let joint_kwargs = Object.assign(defaults_kwargs, kwargs);
+
+  if (units === "IP") {
+    if (body_surface_area === 1.8258) {
```

Comment:
don't set the default value in the parameters list, and detect if they are undefined